src/Agent.Worker/TestResults/NunitResultReader.cs

```diff
@@ -8,46 +8,22 @@
 
 namespace Microsoft.VisualStudio.Services.Agent.Worker.TestResults
 {
-    public class NUnitResultReader : AgentService, IResultReader
+    internal class NUnitResultsXmlReader
     {
-        public Type ExtensionType => typeof(IResultReader);
-        public string Name => "NUnit";
-
-        public NUnitResultReader()
+        public NUnitResultsXmlReader() 
         {
-            AddResultsFileToRunLevelAttachments = true;
         }
 
-        public TestRunData ReadResults(IExecutionContext executionContext, string filePath, TestRunContext runContext = null)
+        public TestRunData GetTestRunData(string filePath, XmlDocument doc, XmlNode testResultsNode, TestRunContext runContext, bool addResultsAsAttachments)
```

Comment:
The function is too big to define it's behavior. Can we refactor it to smaller funcs?